### PR TITLE
Linux pkg sys

### PR DESCRIPTION
**Pull request by klem23**
## Summary
Detecttion  of cuda when system package are used.
Download nvenc Header for linux

## Review
Are you able to compile on linux with system package only except for nvcuvid decoder ?

Closes: #2

### CODE REVIEW

**Comment by jeremad** [comment]
LGTM but I need to be at home to test that

File: IO/src/rtmp/CMakeLists.txt [hunk 1]
```diff
@@ -105,6 +105,23 @@ endif(RTMP_QUICKSYNC)
 
 
 if(RTMP_NVENC)
+
+if(LINUX)
+  include(FetchContent)
+  FetchContent_Declare(
+    nvenc_api
+    GIT_REPOSITORY https://git.videolan.org/git/ffmpeg/nv-codec-headers.git
+    GIT_TAG 106d84a2c1960d1ccac7fa2646f0adc52d2d4ff2
```
Comment (w-m) [inline IO/src/rtmp/CMakeLists.txt]: There seem to be human-readable tags (https://git.videolan.org/?p=ffmpeg/nv-codec-headers.git;a=tags). I'd prefer using one of them over the hash, so the SDK version becomes apparent.

Comment (klem23) [reply IO/src/rtmp/CMakeLists.txt]: Done, it's a good idea, thanks.

Comment (w-m) [reply IO/src/rtmp/CMakeLists.txt]: :+1:

File: IO/src/rtmp/CMakeLists.txt [hunk 2]
```diff
@@ -105,6 +105,23 @@ endif(RTMP_QUICKSYNC)
 
 
 if(RTMP_NVENC)
+
+if(LINUX)
```
Comment (w-m) [inline IO/src/rtmp/CMakeLists.txt]: Just noticed, the whole block needs to be indented by two spaces (it's inside `if(RTMP_NVENC)`).

**Comment by w-m** [comment]
Can't test building locally (no RTMP_NVENC on Mac), I'll leave the review to @jeremad.

File: IO/src/rtmp/CMakeLists.txt (hunk 2 above)
Comment (klem23) [reply IO/src/rtmp/CMakeLists.txt]: Done.

Comment (w-m) [reply IO/src/rtmp/CMakeLists.txt]: 🙇

**Pull request merged**